api/envoy/config/core/v3/health_check.proto

```diff
@@ -103,8 +103,18 @@ message HealthCheck {
     // [#not-implemented-hide:] HTTP specific payload.
     Payload send = 3;
 
-    // [#not-implemented-hide:] HTTP specific response.
-    Payload receive = 4;
+    // Specifies a list of HTTP expected responses to match in the first ``response_buffer_size`` bytes of the response body.
+    // If it is set, both the expected response check and status code determine the health check.
+    // When checking the response, “fuzzy” matching is performed such that each payload block must be found,
+    // and in the order specified, but not necessarily contiguous.
+    //
+    // .. note::
+    //
+    //   Only the first 1024 bytes of the response body is used to match when ``response_buffer_size`` is not set.
+    repeated Payload receive = 4;
+
+    // Specifies the size of response buffer in bytes that is used to Payload match. Default value is 1024.
```

Comment:
IMHO there should be a way to define an "infinite" buffer size, just in case some config wants to avoid setting a defined threshold.

Reply:
Good point. In that case, changing the default behavior to be infinite and getting rid of all 1024 definition will make more sense from user's perspective.